### PR TITLE
Feature/add document management using filefield

### PR DESCRIPTION
**Pull request by 252afh**
## Context

We'd like to use django-storage for managing files, to do this I've changed the model for File in django to use a FileField, and changed the settings in settings.py to configure an S3 connection on localhost and aws

## Changes proposed in this pull request

- Updated setting to use minio S3 and aws S3 on localhost and aws respectively
- Updated views and jinja2 templates to reflect the changes

## Guidance to review

To locally test:
- Use the document management page at `/documents/` to upload a file
- Try the `download` button
- Browse to `localhost:9000` to view minio and check the file exists
- Go back to the document management page, click `delete`
- Check the file is deleted on the document management page and in minio

To test on dev:

- Ask Ell for the AWS credentials and setup for connecting localhost to aws S3 backend

## Things to check

- [x] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [x] I have run integration tests

### CODE REVIEW

File: django_app/pyproject.toml [hunk 1]
```diff
@@ -39,10 +39,10 @@ presidio-analyzer = "^2.2.351"
 en_core_web_sm = {url = "https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-3.7.1/en_core_web_sm-3.7.1.tar.gz"}
 defusedcsv = "^2.0.0"
 python-dotenv = "^1.0.0"
-yarl = "^1.9.4"
 django-allauth = "^0.61.1"
 django-otp = "^1.3.0"
 humanize = "^4.9.0"
+yarl = "^1.9.4"
```
Comment (brunns) [inline django_app/pyproject.toml:45]: I don't believe order matters, so this can be reverted.

File: django_app/redbox_app/redbox_core/models.py [hunk 2]
```diff
@@ -55,22 +54,39 @@ class ProcessingStatusEnum(models.TextChoices):
...
 
     @property
-    def file_type(self):
+    def file_type(self) -> str:
         name = self.original_file.name
         return name.split(".")[-1]
 
     @property
-    def file_url(self):
+    def url(self) -> str:
```
Comment (brunns) [inline django_app/redbox_app/redbox_core/models.py:70]: Could we use a furl.URL here?

**Review by brunns — COMMENTED** [review]
Couple of nitpicks.

File: django_app/redbox_app/redbox_core/models.py (hunk 2 above)
Comment (252afh) [reply django_app/redbox_app/redbox_core/models.py:70]: Changed this to use furl now, I've checked the template still works fine

**Review by brunns — APPROVED** [review]

File: django_app/redbox_app/redbox_core/models.py [hunk 3]
```diff
@@ -55,21 +58,52 @@ class ProcessingStatusEnum(models.TextChoices):
...
+
+            url = s3.generate_presigned_url(
+                ClientMethod="get_object",
+                Params={
+                    "Bucket": settings.AWS_STORAGE_BUCKET_NAME,
+                    "Key": self.name,
+                },
+            )
+            return URL(url)
+        return URL(self.original_file.url)
```
Comment (lmwilkigov) [inline django_app/redbox_app/redbox_core/models.py:96]: Could this have a more explicit if/else flow for readability

**Review by lmwilkigov — APPROVED** [review]
LGTM, minor note on style

**Comment by KevinEtchells** [comment]
@252afh - looks like there's an issue for URLs without a trailing slash that's being introduced. In that `/documents/` works but `/documents` doesn't. Same for `/sessions` - whereas that's working okay on main.

**Pull request merged**